### PR TITLE
922 stepwise hario90

### PR DESCRIPTION
**Pull request by lemillermicrosoft**
### Contribution Checklist

- [ ] The code builds clean without any errors or warnings
- [ ] The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [ ] All unit tests pass, and I have added new tests where possible
- [ ] I didn't break anyone :smile:

### CODE REVIEW

File: dotnet/src/Planners/Planners.Core/Sequential/SKContextSequentialPlannerExtensions.cs [hunk 1]
```diff
@@ -82,23 +83,23 @@ public static class SKContextSequentialPlannerExtensions
...
             // Search for functions that match the semantic query.
-            var memories = config.Memory.SearchAsync(
+            var memories = semanticMemoryConfig.Memory.SearchAsync(
                 PlannerMemoryCollectionName,
                 semanticQuery!,
-                config.MaxRelevantFunctions,
-                config.RelevancyThreshold.Value,
-                cancellationToken: cancellationToken);
+                semanticMemoryConfig.MaxRelevantFunctions,
+                semanticMemoryConfig.RelevancyThreshold.HasValue ? semanticMemoryConfig.RelevancyThreshold.Value : 0.0,
```
Comment (hario90) [inline dotnet/src/Planners/Planners.Core/Sequential/SKContextSequentialPlannerExtensions.cs:130]: good call

**Comment by hario90** [comment]
Looks good so far!

File: dotnet/src/Planners/Planners.Core/Sequential/SKContextSequentialPlannerExtensions.cs (hunk 1 above)
Comment (lemillermicrosoft) [reply dotnet/src/Planners/Planners.Core/Sequential/SKContextSequentialPlannerExtensions.cs:130]: yeah I feel confident about this now that it's in a separate config object, so if you give anything there, that feels like a reasonable default given Memory is available.

File: dotnet/src/Planners/Planners.Core/Utils/FunctionUtils.cs [hunk 2]
```diff
@@ -0,0 +1,11 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+internal static class FunctionUtils
```
Comment (lemillermicrosoft) [inline dotnet/src/Planners/Planners.Core/Utils/FunctionUtils.cs:3]: Did not give much thinking to naming here.

File: dotnet/src/Planners/Planners.Core/Action/ActionPlanner.cs [hunk 3]
```diff
@@ -106,22 +105,21 @@ public async Task<Plan> CreatePlanAsync(string goal, CancellationToken cancellat
...
-            plan = new Plan(goal, this._context.Functions!.GetFunction(parts[0], parts[1]));
-        }
-        else if (!string.IsNullOrWhiteSpace(planData.Plan.Function))
-        {
-            plan = new Plan(goal, this._context.Functions!.GetFunction(planData.Plan.Function));
-        }
-        else
+        Plan? plan = null;
+
+        FunctionUtils.GetFunctionCallbackNames(planData.Plan.Function, out var pluginName, out var functionName);
```
Comment (hario90) [inline dotnet/src/Planners/Planners.Core/Action/ActionPlanner.cs:110]: I would name this something like GetPluginAndFunctionNameFromPlanData

**Pull request closed without merging**